src/clj_kondo/impl/core.clj

```diff
@@ -361,6 +361,47 @@
 (defn classpath? [f]
   (str/includes? f path-separator))
 
+(defn ^:private entries-from-jar [^java.io.File jar-file]
```

Comment:
I think laziness should never escape `with-open`. Also, perhaps we can just call this `entries-from-jar-count`, and already count in there before closing?